### DISCUSSION TITLE
c/c++ 1.4.0 insiders2 version isn't working in workspace

### DISCUSSION THREAD

**Discussion by j0k4rX** [microsoft/vscode-cpptools#7553] [Q&A]
Hi,
recently I installed c/c++ 1.4.0 insiders2 version
and i opened a github repository

but it seems this insiders version doesn't work in my workspace
it works in user though

Can anyone help me fix this problem?

regards

**Comment by Colengms** [comment]
Hi @j0k4rX .  Could you provide more details on what you are seeing not working?  What do you mean by working in 'user', but not in 'workspace' ?

If you enable the following setting:

```
"C_Cpp.loggingLevel": "Debug",
```

That should cause logging to the C/C++ Output channel.  Could you provide that output leading up to the issue?  That might tell us more about what the extension is doing when it's not working.

Also, if you have a `c_cpp_properties.json` file (or `C_Cpp.default.*` settings), could you provide the contents?

**Reply by j0k4rX** [reply]
yes, I only used this `"C_Cpp.updateChannel": "Insiders",` in my user settings.json

the thing is: I am using stable release v1.56.2 of vscode (not insiders version)
when i was editing c++ code, multiple times vscode showed popup like this, "Do you want to use c/c++ insiders version? source: c/c++ extension"
and I just enabled it

Stable release of this extension seems to work fine both in 'user' and 'workspace'
but insiders version doesn't work.
the whole c/cpp extension gets faded in extension area

**Reply by sean-mcmanus** [reply]
I'm not able to repro the issue. It sounds like something is causing installation of the insiders vsix to fail leading it to reprompt. Do you see any related errors in the Help->Toggle Developer Tools->Console logging?

Have you tried manually downloading and installing vsix at https://github.com/microsoft/vscode-cpptools/releases/tag/1.4.0-insiders2 ?

**Reply by Colengms** [reply]
We have occasionally seen issues with VS Code failing to fully install a VSIX, perhaps due to being interrupted by quitting or reloading the window.  Could you try deleting `ms-vscode.cpptools-1.4.0-insiders2` from your `.vscode` directory (in your user directory), and allowing it to be reinstalled?  VS Code will cache that directory and re-use it, so it's necessary to delete it to convince VS code to freshly install it.

**Reply by j0k4rX** [reply]
finally, I got the problem

in the extension area, c/c++ extension is dimmed in color
and it shows this
`this extension has been disabled because it doesn't support virtual workspace`

i tried to enable multiple times through ellipsis but it doesn't work

![Screenshot 2021-05-25 170024](https://user-images.githubusercontent.com/71499522/119521643-2862a280-bd9d-11eb-9b4e-8630bf4bf26a.jpg)

**Reply by sean-mcmanus** [reply]
Yes, our extension doesn't support virtual workspaces.

**Reply by CodElixer** [reply]
> yes, I only used this `"C_Cpp.updateChannel": "Insiders",` in my user settings.json
> 
> the thing is: I am using stable release v1.56.2 of vscode (not insiders version) when i was editing c++ code, multiple times vscode showed popup like this, "Do you want to use c/c++ insiders version? source: c/c++ extension" and I just enabled it
> 
> Stable release of this extension seems to work fine both in 'user' and 'workspace' but insiders version doesn't work. the whole c/cpp extension gets faded in extension area

Hi!
As Insider parameter is showing deprecated for user, So what parameters should be entered normally in the C_Cpp.updateChannel??
You can refer to this post ( https://github.com/microsoft/vscode-cpptools/discussions/9652 ) as here the question is properly formatted!

**Reply by sean-mcmanus** [reply]
You should just remove updateChannel -- we now use the VS Code pre-release mechanism, e.g. https://github.com/microsoft/vscode-cpptools/releases/tag/v1.12.0
